tests/test_edge_cases.py

```diff
@@ -1892,3 +1893,60 @@ class Config:
             arbitrary_types_allowed = True
 
     assert Model().x == Foo()
+
+
+def test_config_partial():
+    class Foo(BaseModel):
+        a: str = Field(...)
+        b: str = Field(...)
+        c: Optional[str] = Field()
+
+    Bar = create_partial_model(Foo)
+
+    # Ensure partial is only set on the new class
+    assert Foo.Config.partial is False
+    assert Bar.Config.partial is True
+
+    # Verify b (a required field) can be omitted on the partial model
+    b = Bar(a='x', c=None)
+    assert b.dict() == {'a': 'x', 'c': None}
+    assert b.dict(exclude_unset=True) == {'a': 'x', 'c': None}
+    assert dict(b) == {'a': 'x', 'c': None}
+
+    # Verify b cannot be passed explicitly as None
+    with pytest.raises(ValidationError) as exc_info:
+        b = Bar(a='x', b=None)
+    assert exc_info.value.errors() == [
+        {'loc': ('b',), 'msg': 'none is not an allowed value', 'type': 'type_error.none.not_allowed'}
+    ]
+
```

Comment:
Can we split this into multiple tests, even if it involves duplicating model definitions.